### ISSUE TITLE
Permissive license or LGPL?

### ISSUE THREAD

**Issue by ericpre** [hyperspy/rosettasciio#51]
Now that the IO code of hyperspy has been split, it may worth considering re-licensing the code with a more permissive license in RosettaSciIO.
I don't a strong view on this topic, but I start this discussion because it is coming up once in a while, and it seems that there is appetite for a more permissive license, as keeping the GPL license could prevent adoption by some libraries/softwares.

**Comment by vasole** [comment]
My preferred license for collaborative work is LGPL: You use it, but if you modify/improve it, you are forced to distribute the modifications.

However, that license was not appropriate for collaboration with some US scientists. Basically their caveats were that if they are paid by the US administration, the US administration must be able to use their code and that could be problematic (DoE, defense,...)

Therefore we have adopted MIT in most projects (hdf5plugin, PyMca, silx, fabIO -purpose somehow related to this project- ...)

**Comment by sem-geologist** [comment]
I think LGPL also is very appropriate license

**Comment by sem-geologist** [comment]
There is other nice features of LGPL. Some of these works (like mine reverse engineering of formats) is to give freedom and save users or comrades-in-occupation (SEM and EPMA operators) from wasting unnecessarily time by thousand of mouse clicks normally required to achieve even basic functions... The usability for other interested parts (OEM, other software developers) is of secondary importance at least for me. Closing such library (or) parts is like undoing the effort of reverse engineering. So LGPL forces to leave such part open, and still gives the freedom to user to replace such library with newer version as LGPL allows only dynamic linking. So the User can use some proprietary software and use open source library with all benefits coming with it. We should remember that LGPL / GPL is to give user a freedom, not to empower some third part OEM. And LGPL I think is this kind of compromise to let it be useful with proprietary software and still maintain huge degree of freedoms for user.

BTW, I can understand what's the problem of DoE defence... Somehow Linux being GPL (mostly RedHat) and running on their War machines going to Iraq was not at all a problem (At least I saw some articles there and here claiming that). What had changed in USA administration/military to get against GPL?

**Comment by vasole** [comment]
Please, do not misunderstand me. If I would have a vote, it would be for LGPL.

| _What had changed in USA administration/military to get against GPL?_

No idea. May be they do not want to get suited because of not distributing their code?

I work for an European research organization and our general software policy is to give access to our developments to the whole community and not only to the countries belonging to the organization.

**Comment by francisco-dlp** [comment]
I agree that LGPL is a more appropriate license for RosettaSciIO (and possibly for HyperSpy too). While it doesn't maximise the potential users of the library (for that we'll have to think on BSD or similar), it increases  substantially its domain of application without compromising on the essential: any modifications to the library must be published under the same terms.

**Comment by sem-geologist** [comment]
One thing what most of different kind of end users does not understand (including mentioned administrations) that no one needs to distribute changes or switch its code base to GPL then using GPL parts - unless they are distributing it for third parties (selling, or giving away attached as a single product). So if DoD (or any other agency) needs own modifications - and for sure they are not going to use those outside from institutions - who ever is going to find out and sue them? The Law which can't be anyhow enforced is a dead law. In my view, Technically I see very little difference between commercial products requiring third part GPL libraries, LGPL libraries or comercial libraries. What the difference between application which requires on windows to get and install some .net runtime compared on linux to get PyQt5, or PySide6, gnuplot, libpng and so on... No difference. The difference is when distributing fully functioning products with all necessary libraries included - and this is where these Licenses are relevant. There are some purists which would want only GPL everything and try to bend the reality to make us believe that we can't mix different licenses (well developers can't, but that is not end users!).

**Comment by sem-geologist** [comment]
an example: `pyqtgraph` library is licensed under BSD. Wait a moment, but PyQt5 requires either GPL or commercial! Yeah, but `pyqtgraph` does not modify the PyQt5 and is not distributing the PyQt5 - which is left to get installed for end users or developers.

**Comment by vasole** [comment]
> It is not that simple: @sem-geologist's code has being edited by a number of other developers. 

But not his very first version:

https://github.com/hyperspy/hyperspy/commit/c198833d7b999a48835508b4c278762afc52cf42

> To distribute the code under a different license that is compatible with your needs we need to ask all the contributors to agree
> changing the license of the files.

For the current status of the code, yes.

**Comment by sem-geologist** [comment]
@vasole tell my why You are so much interested in SFS? SFS is only the first layer (think about it like zip file). SFS allows to extract internal files - it is indeed useful on its own for developing anything else handling Esprit data. i.e. .pan files (where every tile has its own xml file) and EBSD bcf are packed into such SFS containers too. Initially the sfs code was in separate file, but later I had bunched it into same file to make no mess in hyperspy source file hierarchy. Now, as this library is getting the shape I think it would be good idea (independent from the license change or not) to split that code into separate files as it was initially intended as that will improve maintainability.

**Comment by vasole** [comment]
Because I already have tools to parse Bruker XRF files. What I need is access to the internal files of the SFS container.

**Comment by sem-geologist** [comment]
rtx is a different beast! it is (XML based) project file. You still will need binary parser, SFS alone is not enough. Actually the XRF-EDS bcf and SEM-EDS bcf has the same amount of internal files inside sfs container with same file names. The only difference is the Header file (inside SFS container) which for XRF would contain additional nodes describing the X-ray properties, where sample and detector description would be identical to SEM-EDS. Most important binary Cube (which needs lots of decoding and there is this fancy cython code included to parse it really fast) structure is absolutely agnostic if that is electron excited EDS or X-ray excited EDS hyperspectra. You are not first asking for handling XRF (see this #19) and now after solving the #39 maybe it finally can be progressed further (albeit I have no µ-XRF hardware thus it is hard for me to implement it.). But I think You could be useful here as looks You are already familiar what is needed to be read from XML. the Header (XML) file in the SFS container will contain very similar XML nodes as You are familiar already in rtx.

**Comment by vasole** [comment]
By inspecting the file I thought it was just XML embedded in a SFS container and therefore I already had support for the content as I wrote in #50

**Comment by AlexHenderson** [comment]
Personally, I think LGPL is the most restrictive licence that should be applied. My MATLAB toolbox (I know, I know. That was then, but this is now) was licenced GPL, and now I can't change it without throwing out lots of GPL licenced packages I've incorporated. Indeed, I was looking for a Python parser for Renishaw Raman files and came across one licenced GPL. When I asked if they would consider switching to a MIT licence, it turned out they'd modelled their code on my MATLAB version which was GPL. D'oh! We both decided to scale back to MIT for that code, which makes my Python toolkit more attractive. 

In the FAIR data community they say "as open as possible, but as restricted as necessary". What would we say are necessary restrictions? Acknowlegement of original authorship? Requirement that distributed, modified versions are released? Any others? Then determine a licence based on that minimum set. 

For the record, I'm now using MIT.

**Comment by vasole** [comment]
> But I think You could be useful here as looks You are already familiar what is needed to be read from XML. the Header (XML) > file in the SFS container will contain very similar XML nodes as You are familiar already in rtx.

I do not have the equipment. I just provide the analysis software :-)

To be able to load the data, one effectively needs the patch suggested in https://github.com/hyperspy/hyperspy/issues/2898 concerning self.hv 

However, that is not enough with huge maps (the only one I have is 1273 x 1578 x 4096). If I do not down-sample the data, the data type  is wrongly interpreted as uint8 (instead of uint16).

Did anybody encountered that problem already?

**Comment by jlaehne** [comment]
This discussion has somewhat stalled (and partly deviated). Still, I think it would  be good to go about the re-licensing now that we are close to an initial release - even if might only be implemented in a follow-up release. In principle, we had several proposals for different alternative licenses:

- At least, we should move to `LGPL` to be more permissive (and there seems to pretty much a consensus on that)
- For even wider adoption, `MIT` (or `BSD`) might be more suited option(s), but remove the requirement to publish any modifications to the library under the same terms

So as step forward, we should

1. Decide on the licence we aim for (in the next weeks)
2. Try to get agreement from all [Contributors](https://github.com/hyperspy/rosettasciio/graphs/contributors) (there are 38 in total, do we need agreement even from people who have only corrected typos? Any idea how to best get their agreement?)
3. Publish individual readers under separate licenses if we do not get the agreement of everyone

**Comment by CSSFrancis** [comment]
Okay maybe we try to get the ball rolling on this again.  Just starting with the top 10 or so contributors.  

@ericpre @francisco-dlp @jlaehne @sem-geologist @pietsjoh @vidartf @nem1234 @din14970 @ssomnath are you in support of changing to a MIT license?

**Comment by din14970** [comment]
For personal projects that I consider worthless I use MIT. In this case, I am personally not in favor of the MIT license. As I see it, MIT essentially means the code can be used by anyone for any purpose. From personal experience, if the for-profit companies abuse this all the time to take free labor from open source, profit from it in a closed source product, and give nothing back. Even worse, they sometimes start making demands from maintainers. That doesn't sit well with me; I prefer if use of open source (strongly) incentivizes creation of and contribution to open source. If this perspective is short sighted, I'm open to someone changing my mind.

**Comment by jat255** [comment]
In my personal capacity, I'm generally in favor of more open licensing (MIT, BSD, Apache, etc.), perhaps with a clause requiring explicit acknowledgement/citation, since this is very much an academic/community project. I think most of the permissive licenses require that already, actually...

In my work capacity, any contributions are effectively public domain (AFAIU), and thus can be used as such (not a lawyer, obviously). 

---

RE: @din14970's comments:

I agree with your sentiment in general, as there have been many examples of this sort of behavior happening. In contrast, I look to other parts of the Scientific Python ecosystem and notice that most of our "peer" libraries use permissive licensing:

 - [NumPy](https://github.com/numpy/numpy/?tab=License-1-ov-file#readme): BSD
 - [xarray](https://github.com/pydata/xarray?tab=Apache-2.0-1-ov-file#readme): Apache 2.0
 - [astropy](https://github.com/astropy/astropy?tab=BSD-3-Clause-1-ov-file#readme): BSD
 - [napari](https://github.com/Napari/napari?tab=BSD-3-Clause-1-ov-file#readme): BSD
 - [dask](https://github.com/dask/dask#BSD-3-Clause-1-ov-file): BSD
 - [matplotlib](https://github.com/matplotlib/matplotlib/blob/main/LICENSE/LICENSE): Custom permissive license

**Comment by ericpre** [comment]
I agree that this is time to change to a permissive license (MIT or BSD) because there is a need for rosettasciio code to be released with permissive license (for example https://github.com/hyperspy/rosettasciio/issues/51#issuecomment-1273511577 and https://github.com/hyperspy/rosettasciio/issues/314 / https://github.com/LiberTEM/LiberTEM/issues/1649).

Overall, I would except that it would be beneficial for the community to the IO code could be used more easily by third parties, for example:
- it would avoid duplication of IO code (which will happen if we keep the GPL license),
- it would allow other software/libraries to focus on their core functionalities,
- other softwares/libraires depending on rosettasciio are likely to contribute once in a while, to fix or improve what their need

**Comment by CSSFrancis** [comment]
> For personal projects that I consider worthless I use MIT. In this case, I am personally not in favor of the MIT license. As I see it, MIT essentially means the code can be used by anyone for any purpose. From personal experience, if the for-profit companies abuse this all the time to take free labor from open source, profit from it in a closed source product, and give nothing back. Even worse, they sometimes start making demands from maintainers. That doesn't sit well with me; I prefer if use of open source (strongly) incentivizes creation of and contribution to open source. If this perspective is short sighted, I'm open to someone changing my mind.

@din14970 I think that's an appropriate consideration in most cases.  Here, I feel like if a company wants to use this code and make interoperability better than by all means do it.  This is kind of one of those situations where I don't really know if there is money to be made (??although maybe??).  Mostly this is just based on LiberTEM relicensing to MIT and it would be a real shame to not be compatible with them. 

A more permissive license for Hyperspy and pyxem, I think those would require significantly more discusssion.

The point about not giving back... I don't know...  The (maybe too cynical) part of me wants to say that it's not like for profit companies are contributing extensively at the moment. At the very least this might allow me to contribute slightly more as part of my day job.

**Comment by din14970** [comment]
>  In contrast, I look to other parts of the Scientific Python ecosystem and notice that most of our "peer" libraries use permissive licensing

AFAIK (but please correct me if I am wrong on this) this is because many of these projects are funded directly or indirectly (through foundations like NumFOCUS) with corporate money. Corporations have an interest in making OSS licenses permissive.

> https://github.com/hyperspy/rosettasciio/issues/314 / https://github.com/LiberTEM/LiberTEM/issues/1649

Precisely for examples like this is why I would not choose a permissive license. The company will take the code, add a GUI wrapper around it, then sell it back to the same research groups with profit, taking work they didn't have to pay anything for and returning nothing to the original projects. Let's not forget: with GPL they can still use LiberTEM as a dependency, it just means they also have to license their software under GPL which they don't want to do.

> it would avoid duplication of IO code (which will happen if we keep the GPL license)

Honestly I don't really see a problem with that. If you want to sell a closed source project, build your own IO and don't piggy back off of free work from others.

> other softwares/libraires depending on rosettasciio are likely to contribute once in a while, to fix or improve what their need

Having worked in corporate, I can tell you this is very much wishful thinking.

To be clear, I will not battle you on the license and I will not veto the decision that the majority makes. If the majority of people here decide to go with MIT or BSD I will take peace with that. But since you asked my opinion, I am giving it, and it is largely based on the profiteering I have seen at large corporations.

**Comment by nem1234** [comment]
Both licenses have different benefits and disadvantages.
I will follow the general trend.

**Comment by ericpre** [comment]
> Precisely for examples like this is why I would not choose a permissive license. The company will take the code, add a GUI wrapper around it, then sell it back to the same research groups with profit, taking work they didn't have to pay anything for and returning nothing to the original projects. Let's not forget: with GPL they can still use LiberTEM as a dependency, it just means they also have to license their software under GPL which they don't want to do.

A different perspective on that is to think about benefit to the users instead of the project itself: I think that this would be a significant benefit to users if some companies could provide useful user interface of open source tools that integrate well with the instrument itself. This would make it more accessible and facilitate screening of the data during the measurement, for example by using the same (advanced) tools for online and offline analysis easily, etc.

**Comment by CSSFrancis** [comment]
At the end of the day my personal opinion is that supporting something like rosettasciio, hyperspy, pyxem, LiberTem is quite difficult.  With the exception of LiberTem there isn't really sustained monetary support for any of these projects. We are kind of left with a couple of different options.

1. Continue running things as is (and hope @ericpre doesn't get bored)
2. Try to monitize Hyperspy in some way to have more sustained development 
3. Integrate with coorperations and hope for support from coorperate partners
4. Apply for continued funding either through a funding organization or society

I've kind of struggled with this a little bit as I've started to work for one of those coorperations.  Part of my work __is__ supporting open source software with part of my salary being basically donated to help out wherever I would like (although I might spend more time than I'm supposed to). There is also a bit of an open question of "Does this actually gain us customers?", I would like to think so but I'm not entirely sure... I guess we will see.

I guess I will also say it's not like the GPL license is doing us any favors currently.  People still use/ make money off of Hyperspy/ pyxem etc they just don't do it directly.  I'm sure some coorperation uses it for doing data analysis and it's not like they are paying a royalty.

**Comment by jlaehne** [comment]
> A different perspective on that is to think about benefit to the users instead of the project itself: I think that this would be a significant benefit to users if some companies could provide useful user interface of open source tools that integrate well with the instrument itself. This would make it more accessible and facilitate screening of the data during the measurement, for example by using the same (advanced) tools for online and offline analysis easily, etc.

Indeed, intercompatibility between different formats provided by RosettaSciIO does in particular empower the user, because he can choose the tool he prefers for his data analysis. Here, we should aim for the broadest possible applicability, which would mean a permissive license. How permissive we want to be on the end of HyperSpy and the other libraries is another story.

**Comment by din14970** [comment]
Reading and agreeing with the other arguments, I would be fully on board with MIT/BSD if it is paired with some aggressive community outreach & partnership building attempts. Opening up and hoping for the best seems like a squandered opportunity. You risk becoming a low level library that everyone relies on and yet everyone takes for granted and makes demands on; think of OpenBLAS, a fundamental numerical library that pretty much has 1 person working on it. 

Instead, if one actively reaches out and connects with companies that could benefit from the project, one is more proactive in extracting funding/sponsorship and/or developer contribution. This could be done "from the inside" like @CSSFrancis could lobby with their employer or through networks and conferences. Travis Oliphant, the creator of Numpy, has built some interesting models around sustainable open source projects; perhaps we could reach out to him for advice. 

> There is also a bit of an open question of "Does this actually gain us customers?", I would like to think so but I'm not entirely sure... I guess we will see.

Yes, for the people holding the purse strings, there must always be a good business case. One must show at least one of two things: "it increases revenue" or "it reduces costs". Costs are usually easier to calculate than projected revenue increases, and most companies prefer to outsource development for everything that is not their core product, since development and maintenance of software costs a lot and is risky. I think with RosettaSciIO you can sell it more from the cost perspective. Building and maintaining integrations with formats from other vendors is valuable for customers but also expensive if every vendor has to do it on their own. There exist entire businesses on selling "integrations", e.g. Fivetran. Sharing this burden in an open source project is just good business sense.

**Comment by jlaehne** [comment]
> Instead, if one actively reaches out and connects with companies that could benefit from the project, one is more proactive in extracting funding/sponsorship and/or developer contribution.

One lever is also to request compatibility with HyperSpy in tenders, like I have recently done. On the side of CL vendors, Attolight has developed and is supporting their I/O plugin (as customers have requested it, but also as it is a selling point being compatible to HyperSpy!) and we have gotten Delmic to do the same now.

**Comment by sem-geologist** [comment]
As I was previously arguing as Ideally I would see LGPL. If I need to choose from MIT or BSD, I absolutely prefer BSD. Albeit I won't veto MIT.

**Comment by hakonanes** [comment]
Hi all, chiming in here because there is interest (https://github.com/hyperspy/rosettasciio/pull/521) in porting [>10 readers](https://kikuchipy.org/en/stable/tutorials/load_save_data.html#Supported-file-formats) of various EBSD pattern formats from kikuchipy to RosettaSciIO.

I'd prefer a permissive license like BSD-3 or MIT for the readers. They currently carry a GPLv3+ license because kikuchipy depends on HyperSpy. But, I've had plans to re-license them for a while now (which I'm free to do after some minor decoupling from HyperSpy and orix). With a relicensing, I hope for greater adoption of the readers and with that some outside contributors, as I've been the only one developing them (with some very helpful bug reports, of course). There's no guarantee that that will happen, but I consider the potential benefits greater than the potential downsides.

**Comment by ml-evs** [comment]
Perhaps someone could provide an update on the status here? From skimming the discussion it seems there is broad support for relicensing to either LGPL or BSD-3/MIT.

I don't want to re-litigate all that has been discussed above but I will just say that the current GPLv3 renders the library unusable for us over at [datalab-org](https://github.com/datalab-org), forcing us to rely on the initial versions of code that were MIT when contributed (e.g., https://github.com/datalab-org/datalab/pull/1218). This seems to undercut the motivation of this package in the first place.

I couldn't see any specific governance procedure required to make this change (which I guess is the hold-up!) but perhaps one of the core maintainers could chime in as to whether we can expect this change to occur?

**Comment by CSSFrancis** [comment]
Thanks for the nudge @ml-evs...

Full disclosure I had Claude go through this discussion and the contributions to the package to determine the right liscense and what the blockers are. 


## Proposal: BSD-3-Clause

Reading back through the discussion, permissive has broad support (@ericpre, @jlaehne, @jat255, @uellue, @hakonanes, @nem1234, myself, @francisco-dlp) @sem-geologist and @din14970 both prefer otherwise but explicitly said they won't veto. 

Why BSD-3 rather than MIT: it unblocks the same downstream users (LiberTEM and datalab are both MIT; BSD-3 flows into MIT fine), it matches most of our peer ecosystem (numpy, astropy, napari, dask), and it's @sem-geologist's stated preference of the two — "If I need to choose from MIT or BSD, I absolutely prefer BSD". Nobody in the thread preferred MIT over BSD, so this seems like the option with the least friction.

LGPL was the 2022 consensus, but it doesn't actually solve the problem people are hitting and I think with the changing landscape of coding in 2026 with AI BSD makes more sense. 

## Next Steps:

We have no Contributor License Agreemen, so relicensing needs each copyright holder to agree. If you're tagged below, please reply with:

I agree to relicense my contributions to RosettaSciIO under BSD-3-Clause.

### Sign-off tracker

I'll tick these off as replies come in.

**Already indicated support in this thread**

- [x] @ericpre
- [x] @jlaehne
- [x] @jat255
- [x] @francisco-dlp
- [x] @nem1234
- [x] @sem-geologist *(prefers BSD if most support is for permissive  — which is what this proposes)*
- [x] @din14970 *(prefers copyleft, said he won't veto — thank you)*
- [x] @CSSFrancis

**Sign-off still needed** — everyone below owns a substantial or sole-authored reader:

- [x] **@pietsjoh** — `renishaw`, `trivista`, `jobinyvon`, `hamamatsu` (~9,100 lines, sole author of all four). 
- [x] **@Attolight-NTappy** — `digitalsurf`, parts of `tiff` (~2,780) · Attolight
- [x] **@pieleric** / **@noemiebonnet** — `delmic` (~1,650) · Delmic
- [x] **@argerlt** — `topspin` (~830)
- [ ] **@ssomnath** — `usid` (~580) · ORNL
- [x] **@gertjan-devos** — `phenom` (~480) · Thermo Fisher
- [x] **@vidartf** — `blockfile`, `dens`, `utils` (~480)
- [x] **@sempicor** — `jeol`, with @nem1234 (~460)
- [x] **@gguzzina** — `pantarhei` (~460) · CEOS GmbH; the copyright header is corporate, so this should also cover Gregor Mönke's contributions
- [x] **@smribet** — `arina` (~400)
- [x] **@jan-car** — `semper` (~400) · Forschungszentrum Jülich
- [x] **@tjof2** — `utils` + test suite (~340)
- [x] **@DENSmerijn** — `impulse` (~330) · DENSsolutions
- [x] Paul Quinn (Diamond) — `nexus` (~940). Not linked to a GitHub account; I'll email directly. If anyone has a current contact, that would help.
- [x] @to266 
- [x]  @ptim0626 
- [x] @magnunor 
- [ ]  @pburdet 
- [x]  @TomSlater 
- [x]  @arnduveg 
- [x] @k8macarthur 
- [x]  @edwinsupple 
- [x]  @maclariz

Timeline: I'll leave this open for ~1 week and make the change on Monday 31 August if there are no objections. If there are parts that need to be kept GPL I'll seperate them into a clearly marked subpackage. And if you have a concern you want discussed rather than a flat no, please raise it as well. 

If you contributed a typo fix, a CI tweak, or a handful of lines, you don't need to do anything — those fall below the threshold where this matters, and I'd rather not spam ~45 people over 377 lines in total.

## Housekeeping I'll handle regardless
A few files have third-party provenance that needs sorting out independently of anyone's vote. I'll open a separate PR for these:

- [ ] quantumdetector/_api.py — header says it's adapted from [matkraj/read_mib](https://github.com/matkraj/read_mib), which is GPL-3.0. @matkraj are you okay with BSD?  I'm not sure how much of that code is left but if you're okay with the liscense change that makes my life a bit easier :) 

- [ ] utils/_tests.py — assert_deep_almost_equal is credited to oq-engine, which is AGPL-3.0. I'll just rewrite it.

- [x] digitalmicrograph/_api.py and utils/_readfile.py — descend from Stefano Mazzucco's dm3 plugin. No lines of his survive, and _readfile.py still carries a GPL-v2+ header that contradicts the rest of the repo. Worth cleaning up either way.

- [ ] renishaw/_api.py incorporates py-wdf-reader, which is MIT — compatible, notice stays as-is.

To be explicit — git history stays as it is. No commits get rewritten. This only changes the license going forward, same as numpy, matplotlib and LiberTEM did.

**Comment by argerlt** [comment]
I agree to relicense my contributions to RosettaSciIO under BSD-3-Clause.

Thank you for tackling this. Internally, contributing to BSD over GPL is looked upon much more favorably at my lab as well, this will make it much easier to convince people to align efforts with hyperspy.

**Comment by jlaehne** [comment]
As I supervised @pietsjoh to make the readers he wrote available for our lab, I would chip in that it is OK to relicense his contributions to BSD-3 from my point of view. An explicit opt-in from him would of course still be nice, but if there is no response within the indicated timeframe it should not hold up the process.

**Comment by jan-car** [comment]
From my side there are also no objections to relicense under BSD-3-Clause 👍

**Comment by smribet** [comment]
I agree to relicense my contributions to RosettaSciIO under BSD-3-Clause.

**Comment by pieleric** [comment]
Hi, on behalf of Delmic, I and Noémie Bonnet agree to relicense the `delmic` importer (ie, all our contributions)  to RosettaSciIO under BSD-3-Clause.

**Comment by Attolight-NTappy** [comment]
Hi everyone,

no objections to relicense under BSD-3

Cheers

Nicolas

**Comment by DENSmerijn** [comment]
Hi everyone,

On behalf of DENSsolutions, I agree to relicense my contributions to RosettaSciIO under BSD-3-Clause.

Kind regards,
Merijn

**Comment by gertjan-devos** [comment]
Hi everyone,

I have no objections to relicense my RosettaSciIO contribution under BSD-3.

Regards,

Gert-Jan

**Comment by vidartf** [comment]
> Full disclosure I had Claude go through this discussion and the contributions to the package to determine the right liscense and what the blockers are.

It seems like Claude missed some contributors? Could you instead ask it to write a script for deterministically tracing a file and its contributors across renames and moves? I looked at the blockfile reader, and it seems to at least be missing @to266 .

Other than that, I'm ok with relicencing my contributions to the IO code under BSD-3.

**Comment by jlaehne** [comment]
Indeed there seem to be a few more people in the [contributors list](https://github.com/hyperspy/rosettasciio/graphs/contributors?all=1) with more than 500 lines of code contributed like @ptim0626, @magnunor, @pburdet, @TomSlater, @arnduveg, @k8macarthur, @edwinsupple, @maclariz

**Comment by arnduveg** [comment]
Hi everyone,

Indeed, my former username when I contributed to this project was @sempicor which was listed by Claude.
I also have no objection to my contributions being relicenced under BSD-3.

Cheers and thanks for all your work !

**Comment by maclariz** [comment]
Very happy to see this relicensed to BSD-3

**Comment by CSSFrancis** [comment]
> It seems like Claude missed some contributors? Could you instead ask it to write a script for deterministically tracing a file and its contributors across renames and moves? I looked at the blockfile reader, and it seems to at least be missing [@to266](https://github.com/to266) .
> 
> Other than that, I'm ok with relicencing my contributions to the IO code under BSD-3.


Eeek... Was trying to do this quickly and see where that got me... I'll add the additional contributors to the list.  Thanks @vidartf!

**Comment by edwinsupple** [comment]
I agree to relicense my contributions under BSD-3.

**Comment by ptim0626** [comment]
I agree to relicense my contributions to RosettaSciIO under BSD-3-Clause.

**Comment by ericpre** [comment]
Thank you @CSSFrancis for re-activating this issue! :)

From a quick look at the list, I would expect @matao1984 and @AndrewHerzing to be on the list. Could it be that the threshold used by Claude to define "contribution that's matter" needs to be tweaked?

**Comment by magnunor** [comment]
I am fine with relicensing my contributions as BSD-3.

**Comment by vidartf** [comment]
> Indeed there seem to be a few more people in the [contributors list](https://github.com/hyperspy/rosettasciio/graphs/contributors?all=1) with more than 500 lines of code contributed like ...

I'm not really planning to use this under BSD, but for my future reference: is there some magic cutoff at 500 lines of code? Or is the idea that if you filter out the small contributions, it shouldn't be too hard to fill the gap?

**Comment by CSSFrancis** [comment]
> > Indeed there seem to be a few more people in the [contributors list](https://github.com/hyperspy/rosettasciio/graphs/contributors?all=1) with more than 500 lines of code contributed like ...
> 
> I'm not really planning to use this under BSD, but for my future reference: is there some magic cutoff at 500 lines of code? Or is the idea that if you filter out the small contributions, it shouldn't be too hard to fill the gap?

I guess I'm mostly just trying to walk a fine line of not stepping on anyone's toes and trying to get some movement on this issue.  I guess as long as there is general support for this then I can handle the small bits and pieces. Likely that means a more thorough pass to see if anyone was missed. I think from a licensing standpoint each contributor owns their unique and copy writeable code. What clears the bar for unique and copy writable is maybe a bit up to debate, but I'd say that simple bugfixes/ spelling/documentation don't quite clear the bar where full readers and implementations for additional features do. That isn't to say those contributions aren't valuable but there isn't a way to "undo" a contribution that makes something less correct, whereas for a reader/a feature I can explicitly separate and bisect the project to account for that.

I'm hoping this isn't coming off too pushy :) The field obviously is interested in the project, but the licensing is definitely a problem and there seems like general support for this.  If someone has general concerns, I'm happy to discuss them, but I think we kind of have to realistic and realize that without this licensing change then a large portion of this work is going to be duplicated and moved away from this project. 

That being said a couple of more who would be worth getting comments/ a sign off from:

 @msarahan @dnjohnstone I was wondering if you could potentially sign off on this as well?  Or if you have any comments/ concerns I'd love to hear them

**Comment by msarahan** [comment]
Fine by me

**Comment by TomSlater** [comment]
All okay for me.

**Comment by tjof2** [comment]
No concerns

**Comment by CSSFrancis** [comment]
Okay mostly there. These are the last set of contributors that I'd like to hear back from. I've sent out some emails and will continue to update the list.  I might wait till after IMC this week as people might be traveling.  @francisco-dlp if you are at IMC and see some of these people could you ping them in person to give a nod of approval?  Thanks!

- [ ] @ssomnath 
- [x] @gguzzina
- [x] @to266 Does someone have a way to contact Tomas? 
- [x] @AndrewHerzing 
- [x] @matao1984
- [x] @PauTorru 
- [x] @robbmcleod 
- [x] @k8macarthur (confirmed via email)
- [x] @matkraj
- [x] @thomasaarholt

**Comment by AndrewHerzing** [comment]
OK by me.

**Comment by matao1984** [comment]
OK for me.

**Comment by PauTorru** [comment]
Ok by me too!

**Comment by to266** [comment]
I'm ok with this

**Comment by gguzzina** [comment]
Sorry for the delay, both @tensiohead and I are out of office.
This is ok with us.

**Comment by matkraj** [comment]
All good with me too!

**Comment by CSSFrancis** [comment]
Thanks everyone!  I think we are mostly agreed to change the license.  I'm going to make a PR and hope to hear back from @ssomnath and @robbmcleod. Otherwise I'll start scoping what their additions are and if we can move forward without them!

**Comment by robbmcleod** [comment]
My preference is BSD 3-clause, LGPL or Apache-2.0 are both fine too.

**Comment by jlaehne** [comment]
> Thanks everyone! I think we are mostly agreed to change the license. I'm going to make a PR and hope to hear back from [@ssomnath](https://github.com/ssomnath) and [@robbmcleod](https://github.com/robbmcleod). Otherwise I'll start scoping what their additions are and if we can move forward without them!

@pburdet is also open in the list.

@ssomnath was working for Oak Ridge Lab when contributing, and what I have seen from the discussion above they usually seem to prefer the more open licenses.

**Comment by jlaehne** [comment]
I think @thomasaarholt is missing in the list, and he was quite active for some time!

**Comment by thomasaarholt** [comment]
I agree to relicense my contributions to RosettaSciIO under BSD-3-Clause.

Thanks Jonas!
